docs/INITIALIZING.md

```diff
@@ -51,6 +51,12 @@ The store itself defaults to `$HOME/.tpm2_pkcs11` unless specified via the envir
 
   Their is no requirement to use the simulator and abrmd, this is all configuration dependent.
 
+**LOCKING**
+
+When the SQL database is on the disk, the lock is set within the same folder than the SQL file.
+It can lead to some issues if the lock is not released (system crash, reboot), mostly on embedded
+systems. Another folder, for instance a tmpfs one, can be enforced using the env `PKCS11_SQL_LOCK=/var/run/pkcs11_sql_locks`.
```

Comment:
What happens if their are N instances of the library running? We would need N distinct lock files. I am ok with moving it to a tmpfs by default and not needing an env variable for it.

Reply:
Same as using the SQL file: if the instances share the same SQL file so they should share the same lock path.

Reply:
Please, @williamcroberts , are you maintainer ? If so, can you unlock the workflow ?

Reply:
> Same as using the SQL file: if the instances share the same SQL file so they should share the same lock path.

So `take_lock()` takes in the store path with db filename, like `/home/bill/mystore/tpm2_pkcs11.sqlite3` and appends `.lock`, so you end up with a `/home/bill/mystore/tpm2_pkcs11.sqlite3.lock`, one lock file per store. Stores are controlled by default searching and an env variable. But in no way could you map a store to a different lock file.

In this case, its possible to get us to a state where you could have store X, Y and Z in use but have them all pointed to one lock file or have store X in use and point it to two different lock files.

You could probably swap all path separators `/` with underscores `_` and then create a file, using my previous full example of `/var/run/pkcs11_locks/home_bill_mystore_tpm2_pkcs11.sqlite3.lock`. Then you would have the same mapping of 1 lock to 1 store.

Reply:
> Please, @williamcroberts , are you maintainer ? If so, can you unlock the workflow ?

Sure, but lets get those changes in and ill approve it.

Reply:
ok, got it, you are right.
See the new update 59404ca0db6d646532e602fcb30bf5b456239918 that does the substitution of '/' by '_'/

Reply:
done